djconnectwise/sync.py

```diff
@@ -943,27 +956,19 @@ def _assign_field_data(self, instance, json_data):
                 models.Member.objects.filter(id__in=json_data.get('members'))
             )
 
-        self.m2m_changed = self._m2m_has_changed(instance, members)
+        instance_members = list(instance.members.all())
+        self.m2m_changed = self._m2m_has_changed(instance_members, members)
         if self.m2m_changed:
```

Comment:
Great idea to move that behavior into a mixin, but there are some new problems now. Setting an attribute on self from a method on itself is a bit of a red flag, but it wasn't a deal breaker when it was just on the Team synchronizer. But now that `_m2m_has_changed` is on a mixin and `self.m2m_changed` is being set in multiple places, it is a problem.

We don't want to trust each class inheriting from `M2MAssignmentMixin` to correctly set `m2m_changed`. It would be better if `_m2m_has_changed` set `self.m2m_changed` inside the method when it is run. That way child classes only have to call `_m2m_has_changed` to test if that field has changed. Also, `_m2m_has_changed` should be changed to `m2m_has_changed`. Since its being called by child classes, its just best practice to not have it as a private method.

This pattern isn't very robust though, it works right now because we don't have any models with more than one m2m field, but as soon as that changes it will need to be updated. But, who knows how long that day will take to come. Maybe make a comment on the `M2MAssignmentMixin` that mentions this only works if the model has one m2m field.

Reply:
Hi Sam, /wo setting `self.m2m_changed` in a Mixin, we can use `M2MAssignmentMixin` to set multiple m2m fields, which was considered distinguish setting and deciding if m2m fields are changed. Do you want setting the field in a Mixin and reduce that only 1 m2m filed?

Reply:
`self.m2m_changed` is just a boolean value that is set regardless of the fields its checking. Setting it in `_m2m_has_changed` should not reduce how many m2m fields it can be used for.

Reply:
Ready now~